homeassistant/components/shelly/utils.py

```diff
@@ -389,6 +389,8 @@ def device_update_info(
         identifiers={(DOMAIN, entry.entry_id)},
         connections={(CONNECTION_NETWORK_MAC, format_mac(entry.unique_id))},
     ):
+        if device.sw_version == shellydevice.firmware_version:
```

Comment:
My intention in my previous comment was that we do this comparison before we call `device_update_info`, but doing the check inside this method is better, but I would move it to be the first thing (we can supply the current FW as a parameter), before we try to get the device from the registry and we can also remove the check from gen1 after we do it here (now we have a double check for Gen1)